### DISCUSSION TITLE
Searching on system settings

### DISCUSSION THREAD

**Discussion by bertdida** [davatorium/rofi#1670] [Ideas/Feature Requests]
Hi, does anyone here thought about using rofi to search-then-open system settings? I'm curious if this is possible.

**Comment by DaveDavenport** [comment]
No idea , don't have a clue what you mean with 'system settings'. 

You need to provide way more information is you want other people to understand.
(distribution? desktop environment? etc.)

**Reply by bertdida** [reply]
Oh sorry for not being clear. By "system settings" I mean the Control Center.
![Screenshot from 2022-08-07 15-16-45](https://user-images.githubusercontent.com/38489033/183280042-050b9171-2262-4fa3-a910-f6bc0e0413a7.png)

Because on Fedora (which I'm using), settings are included on the search result. I wonder if someone was able to do it, or if this is possible.
![Screenshot from 2022-08-07 15-22-15](https://user-images.githubusercontent.com/38489033/183280103-944db63d-c35e-4dc5-b2fa-daa9d371d844.png)

I'm using Fedora 36, Gnome, x11 btw. thank you

**Reply by DaveDavenport** [reply]
no idea how it is implemented? if they follow a standard, I expect the settings entries to have a desktop file?

**Reply by bertdida** [reply]
Yes they do. Below their desktop files are prefixed with `gnome-`.

![image](https://user-images.githubusercontent.com/38489033/183283320-bbe2dc0c-61f4-4b79-84a1-d5b3349a1c96.png)


Checking on one of the file it contains the metadata below.
```
[Desktop Entry]
Name=Displays
Comment=Choose how to use connected monitors and projectors
Exec=gnome-control-center display
# Translators: Do NOT translate or transliterate this text (this is an icon file name)!
Icon=org.gnome.Settings-display-symbolic
Terminal=false
Type=Application
NoDisplay=true
StartupNotify=true
Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
OnlyShowIn=GNOME;Unity;
X-GNOME-Bugzilla-Bugzilla=GNOME
X-GNOME-Bugzilla-Product=gnome-control-center
X-GNOME-Bugzilla-Component=Screen resolution
X-GNOME-Bugzilla-Version=42.3
Keywords=Panel;Projector;xrandr;Screen;Resolution;Refresh;Monitor;Night;Light;Blue;redshift;color;sunset;sunrise;
```

Knowing that they use desktop files, do you think what I want to achieve is possible? If so, can you suggest me how.

**Reply by DaveDavenport** [reply]
This is why rofi is not displaying them:
```
NoDisplay=true
```
See: https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html

**Reply by bertdida** [reply]
Oh okay. Thanks

**Comment by bertdida** [comment]
To anyone interested in this, I created its plugin https://github.com/bertdida/rofi-control-center. 🥂

**Reply by DaveDavenport** [reply]
Great. This is exactly why we have script and combi mode :smile:
